Prefabs/Pointers/SharedResources/Scripts/PointerFacadeComponentGameObjectExtractor.cs

```diff
@@ -0,0 +1,119 @@
+        /// The source to extract from.
+        /// </summary>
+        [Serialized, Cleared]
+        [field: DocumentedByXml]
+        public PointerFacade Source { get; set; }
+        /// <summary>
+        /// The <see cref="PointerComponentType"/> to extract from the <see cref="Source"/>.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public PointerComponentType PointerComponent { get; set; } = PointerComponentType.Caster;
+
+        /// <inheritdoc />
+        public override GameObject Extract()
+        {
+            Result = null;
+
+            if (Source == null)
+            {
+                return null;
+            }
+
+            switch (PointerComponent)
+            {
+                case PointerComponentType.Caster:
+                    Result = Source.Configuration.Caster.gameObject;
+                    break;
+                case PointerComponentType.PointerElementOrigin:
+                    Result = Source.Configuration.ObjectPointer.Origin.gameObject;
+                    break;
+                case PointerComponentType.PointerElementRepeatedSegment:
+                    Result = Source.Configuration.ObjectPointer.RepeatedSegment.gameObject;
+                    break;
+                case PointerComponentType.PointerElementDestination:
+                    Result = Source.Configuration.ObjectPointer.Destination.gameObject;
+                    break;
+                default:
+                    return null;
+            }
+
+            return base.Extract();
+        }
+
+        /// <summary>
+        /// Extracts the <see cref="Source"/> <see cref="GameObject"/> from the given <see cref="PointerFacade"/> data.
+        /// </summary>
+        /// <param name="data">The <see cref="PointerFacade"/> payload data to extract from.</param>
+        /// <returns>The <see cref="Source"/> <see cref="GameObject"/> within the <see cref="PointerFacade"/>.</returns>
+        public virtual GameObject Extract(PointerFacade facade)
+        {
+            Source = facade;
+            return Extract();
+        }
+
+        /// <summary>
+        /// Extracts the <see cref="Source"/> <see cref="GameObject"/> from the given <see cref="PointerFacade"/> data.
+        /// </summary>
+        /// <param name="data">The <see cref="PointerFacade"/> payload data to extract from.</param>
+        public virtual void DoExtract(PointerFacade facade)
+        {
+            Extract(facade);
+        }
+
+        /// <summary>
+        /// Sets the <see cref="Source"/> based on a given <see cref="GameObject"/>.
+        /// </summary>
+        /// <param name="source">The data that contains the <see cref="PointerFacade"/> component.</param>
+        [RequiresBehaviourState]
+        public virtual void SetSource(GameObject source)
+        {
+            if (source == null)
+            {
+                return;
+            }
+
+            Source = source.GetComponent<PointerFacade>();
```

Comment:
Use our extension method `TryGetComponent` here to work around the editor garbage generation in case that component doesn't exist.

If we apply this suggestion I think we can also get rid of the `if` in this method body as the extension method handles calling it on a `null` object!